serial/src/main/java/org/jboss/marshalling/serial/SerialMarshaller.java

```diff
@@ -328,6 +330,42 @@ private SerialObjectOutputStream getObjectOutputStream() throws IOException {
         return oos;
     }
 
+    private void doWriteRecord(final SerializableClass info, final Object obj) throws IOException {
+        for (SerializableField serializableField : info.getFields()) {
+            switch (serializableField.getKind()) {
+                case BOOLEAN:
+                    writeBoolean((boolean) serializableField.getRecordComponentValue(obj));
+                    break;
+                case BYTE:
+                    writeByte((byte) serializableField.getRecordComponentValue(obj));
+                    break;
+                case SHORT:
+                    writeShort((short) serializableField.getRecordComponentValue(obj));
+                    break;
+                case INT:
+                    writeInt((int) serializableField.getRecordComponentValue(obj));
+                    break;
+                case CHAR:
+                    writeChar((char) serializableField.getRecordComponentValue(obj));
+                    break;
+                case LONG:
+                    writeLong((long) serializableField.getRecordComponentValue(obj));
+                    break;
+                case DOUBLE:
+                    writeDouble((double) serializableField.getRecordComponentValue(obj));
+                    break;
+                case FLOAT:
+                    writeFloat((float) serializableField.getRecordComponentValue(obj));
+                    break;
+            }
+        }
```

Comment:
Why here are two enhanced for loops? Those could be merged into one or am I missing something?

Reply:
Thanks for clarification Ricardo.